File: android/api-tests/src/test/java/com/revenuecat/apitests/java/CommonApiTests.java
package com.revenuecat.apitests.java;

import android.app.Activity;
import android.content.Context;

import androidx.annotation.NonNull;

import com.revenuecat.purchases.DangerousSettings;
import com.revenuecat.purchases.PurchasesAreCompletedBy;
import com.revenuecat.purchases.Store;
import com.revenuecat.purchases.common.PlatformInfo;
import com.revenuecat.purchases.hybridcommon.CommonKt;
import com.revenuecat.purchases.hybridcommon.ErrorContainer;
import com.revenuecat.purchases.hybridcommon.OnResult;
import com.revenuecat.purchases.hybridcommon.OnResultAny;
import com.revenuecat.purchases.hybridcommon.OnResultList;
import com.revenuecat.purchases.models.InAppMessageType;

import java.util.List;
import java.util.Map;

import kotlin.Unit;
import kotlin.jvm.functions.Function1;

@SuppressWarnings({"unused", "deprecation"})
class CommonApiTests {
    private void checkCheckSetAllowSharingAppStoreAccount(boolean enabled) {
        CommonKt.setAllowSharingAppStoreAccount(enabled);
    }

    private void checkGetOfferings(OnResult onResult) {
        CommonKt.getOfferings(onResult);
    }

    private void checkGetProductInfo(List<String> productIDs,
                                     String type,
                                     OnResultList onResult) {
        CommonKt.getProductInfo(productIDs, type, onResult);
    }

    private void checkPurchaseProduct(Activity activity,
                                      String productIdentifier,
                                      String type,
                                      String googleBasePlanId,
                                      String googleOldProductId,
                                      Integer googleReplacementMode,
                                      Boolean googleIsPersonalizedPrice,
                                      Map<String, Object> presentedOfferingContext,
                                      OnResult onResult,
                                      List<Map<String, Object>> addOnStoreProducts,
                                      List<Map<String, Object>> addOnSubscriptionOptions,
                                      List<Map<String, Object>> addOnPackages) {
        CommonKt.purchaseProduct(
                activity,
                productIdentifier,
                type,
                googleBasePlanId,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult
        );

        CommonKt.purchaseProduct(
                activity,
                productIdentifier,
                type,
                googleBasePlanId,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts
        );

        CommonKt.purchaseProduct(
                activity,
                productIdentifier,
                type,
                googleBasePlanId,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions
        );

        CommonKt.purchaseProduct(
                activity,
                productIdentifier,
                type,
                googleBasePlanId,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions,
                addOnPackages
        );
    }

    private void checkPurchasePackage(Activity activity,
                                      String packageIdentifier,
                                      Map<String, Object> presentedOfferingContext,
                                      String googleOldProductId,
                                      Integer googleReplacementMode,
                                      Boolean googleIsPersonalizedPrice,
                                      OnResult onResult,
                                      List<Map<String, Object>> addOnStoreProducts,
                                      List<Map<String, Object>> addOnSubscriptionOptions,
                                      List<Map<String, Object>> addOnPackages) {
        CommonKt.purchasePackage(
                activity,
                packageIdentifier,
                presentedOfferingContext,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                onResult
        );

        CommonKt.purchasePackage(
                activity,
                packageIdentifier,
                presentedOfferingContext,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                onResult,
                addOnStoreProducts
        );

        CommonKt.purchasePackage(
                activity,
                packageIdentifier,
                presentedOfferingContext,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions
        );

        CommonKt.purchasePackage(
                activity,
                packageIdentifier,
                presentedOfferingContext,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions,
                addOnPackages
        );
    }

    private void checkPurchaseSubscriptionOption(Activity activity,
                                                 String productIdentifier,
                                                 String optionIdentifier,
                                                 String googleOldProductId,
                                                 Integer googleReplacementMode,
                                                 Boolean googleIsPersonalizedPrice,
                                                 Map<String, Object> presentedOfferingContext,
                                                 OnResult onResult,
                                                 List<Map<String, Object>> addOnStoreProducts,
                                                 List<Map<String, Object>> addOnSubscriptionOptions,
                                                 List<Map<String, Object>> addOnPackages) {
        CommonKt.purchaseSubscriptionOption(
                activity,
                productIdentifier,
                optionIdentifier,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult
        );

        CommonKt.purchaseSubscriptionOption(
                activity,
                productIdentifier,
                optionIdentifier,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts
        );

        CommonKt.purchaseSubscriptionOption(
                activity,
                productIdentifier,
                optionIdentifier,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions
        );

        CommonKt.purchaseSubscriptionOption(
                activity,
                productIdentifier,
                optionIdentifier,
                googleOldProductId,
                googleReplacementMode,
                googleIsPersonalizedPrice,
                presentedOfferingContext,
                onResult,
                addOnStoreProducts,
                addOnSubscriptionOptions,
                addOnPackages
        );
    }

    private void checkGetStorefront() {
        CommonKt.getStorefront(new Function1<Map<String, ? extends Object>, Unit>() {
            @Override
            public Unit invoke(Map<String, ?> stringMap) {
                return null;
            }
        });
        CommonKt.getStorefront(stringMap -> null);
    }

    private void checkGetAppUserId() {
        String appUserId = CommonKt.getAppUserID();
    }

    private void checkRestorePurchases(OnResult onResult) {
        CommonKt.restorePurchases(onResult);
    }

    private void checkLogIn(String appUserId, OnResult onResult) {
        CommonKt.logIn(appUserId, onResult);
    }

    private void checkLogOut(OnResult onResult) {
        CommonKt.logOut(onResult);
    }

    private void checkSetDebugLogsEnabled(boolean enabled) {
        CommonKt.setDebugLogsEnabled(enabled);
    }

    private void checkSetLogLevel(String level) {
        CommonKt.setLogLevel(level);
    }

    private void checkSetLogHandler(Function1<? super Map<String, String>, Unit> callback) {
        CommonKt.setLogHandler(callback);
    }

    private void checkSetLogHandlerWithOnResult() {
        CommonKt.setLogHandlerWithOnResult(new OnResult() {
            @Override
            public void onReceived(@NonNull Map<String, ?> map) {
            }

            @Override
            public void onError(@NonNull ErrorContainer errorContainer) {
            }
        });
    }

    private void checkSetProxyURLString(String proxyUrlString) {
        CommonKt.setProxyURLString(proxyUrlString);
    }

    private void checkGetProxyURLString() {
        String proxyUrlString = CommonKt.getProxyURLString();
    }

    private void checkGetCustomerInfo(OnResult onResult) {
        CommonKt.getCustomerInfo(onResult);
    }

    private void checkSyncPurchases() {
        CommonKt.syncPurchases();
    }

    private void checkIsAnonymous() {
        boolean isAnonymous = CommonKt.isAnonymous();
    }

    private void checkSetPurchasesAreCompletedBy(String purchasesAreCompletedBy) {
        CommonKt.setPurchasesAreCompletedBy(purchasesAreCompletedBy);
    }

    private void checkCheckTrialOrIntroductoryPriceEligibility(List<String> productIdentifiers) {
        Map<String, Map<String, Object>> result = CommonKt.checkTrialOrIntroductoryPriceEligibility(productIdentifiers);
    }

    private void checkInvalidateCustomerInfoCache() {
        CommonKt.invalidateCustomerInfoCache();
    }

    private void checkCanMakePayments(Context context, List<Integer> features, OnResultAny<Boolean> onResult) {
        CommonKt.canMakePayments(context, features, onResult);
    }

    private void checkShowInAppMessagesIfNeeded(Activity activity, List<InAppMessageType> types) {
        CommonKt.showInAppMessagesIfNeeded(activity);
        CommonKt.showInAppMessagesIfNeeded(activity, types);
        CommonKt.showInAppMessagesIfNeeded(activity, null);
    }

    private void checkConfigure(Context context,
                                String apiKey,
                                String appUserId,
                                String purchasesAreCompletedBy,
                                PlatformInfo platformInfo,
                                Store store,
                                DangerousSettings dangerousSettings,
                                Boolean shouldShowInAppMessagesAutomatically,
                                String verificationMode,
                                Boolean pendingTransactionsForPrepaidPlansEnabled,
                                Boolean diagnosticsEnabled
    ) {
        CommonKt.configure(context, apiKey, appUserId, purchasesAreCompletedBy, platformInfo);
        CommonKt.configure(context, apiKey, appUserId, purchasesAreCompletedBy, platformInfo, store);
        CommonKt.configure(context, apiKey, appUserId, purchasesAreCompletedBy, platformInfo, store, dangerousSettings);
        CommonKt.configure(
                context,
                apiKey,
                appUserId,
                purchasesAreCompletedBy,
                platformInfo,
                store,
                dangerousSettings,
                shouldShowInAppMessagesAutomatically);
        CommonKt.configure(
                context,
                apiKey,
                appUserId,
                purchasesAreCompletedBy,
                platformInfo,
                store,
                dangerousSettings,
                shouldShowInAppMessagesAutomatically,
                verificationMode,
                pendingTransactionsForPrepaidPlansEnabled,
                diagnosticsEnabled);
    }

    private void checkGetPromotionalOffer() {
        ErrorContainer errorContainer = CommonKt.getPromotionalOffer();
    }

    private void checkGetVirtualCurrencies(OnResult onResult) {
        CommonKt.getVirtualCurrencies(onResult);
    }

    private void checkInvalidateVirtualCurrenciesCache() {
        CommonKt.invalidateVirtualCurrenciesCache();
    }

    private void checkGetCachedVirtualCurrencies() {
        Map<String, Object> cachedVirtualCurrencies = CommonKt.getCachedVirtualCurrencies();
    }

    private void checkErrorContainer(Integer code, String message, Map<String, Object> info) {
        ErrorContainer errorContainer = new ErrorContainer(code, message, info);
        Integer storedCode = errorContainer.getCode();
        String storedMessage = errorContainer.getMessage();
        Map<String, Object> storedAny = errorContainer.getInfo();
    }

    private void checkOverridePreferredLocale(String locale) {
        CommonKt.overridePreferredLocale(locale);
    }
}

File: android/api-tests/src/test/java/com/revenuecat/apitests/kotlin/CommonApiTests.kt
package com.revenuecat.apitests.kotlin

import android.app.Activity
import android.content.Context
import com.revenuecat.purchases.DangerousSettings
import com.revenuecat.purchases.Store
import com.revenuecat.purchases.common.PlatformInfo
import com.revenuecat.purchases.hybridcommon.ErrorContainer
import com.revenuecat.purchases.hybridcommon.OnResult
import com.revenuecat.purchases.hybridcommon.OnResultAny
import com.revenuecat.purchases.hybridcommon.OnResultList
import com.revenuecat.purchases.hybridcommon.canMakePayments
import com.revenuecat.purchases.hybridcommon.checkTrialOrIntroductoryPriceEligibility
import com.revenuecat.purchases.hybridcommon.configure
import com.revenuecat.purchases.hybridcommon.getAppUserID
import com.revenuecat.purchases.hybridcommon.getCachedVirtualCurrencies
import com.revenuecat.purchases.hybridcommon.getCustomerInfo
import com.revenuecat.purchases.hybridcommon.getOfferings
import com.revenuecat.purchases.hybridcommon.getProductInfo
import com.revenuecat.purchases.hybridcommon.getPromotionalOffer
import com.revenuecat.purchases.hybridcommon.getProxyURLString
import com.revenuecat.purchases.hybridcommon.getStorefront
import com.revenuecat.purchases.hybridcommon.getVirtualCurrencies
import com.revenuecat.purchases.hybridcommon.invalidateCustomerInfoCache
import com.revenuecat.purchases.hybridcommon.invalidateVirtualCurrenciesCache
import com.revenuecat.purchases.hybridcommon.isAnonymous
import com.revenuecat.purchases.hybridcommon.logIn
import com.revenuecat.purchases.hybridcommon.logOut
import com.revenuecat.purchases.hybridcommon.overridePreferredLocale
import com.revenuecat.purchases.hybridcommon.purchase
import com.revenuecat.purchases.hybridcommon.purchasePackage
import com.revenuecat.purchases.hybridcommon.purchaseProduct
import com.revenuecat.purchases.hybridcommon.purchaseSubscriptionOption
import com.revenuecat.purchases.hybridcommon.restorePurchases
import com.revenuecat.purchases.hybridcommon.setAllowSharingAppStoreAccount
import com.revenuecat.purchases.hybridcommon.setDebugLogsEnabled
import com.revenuecat.purchases.hybridcommon.setLogHandler
import com.revenuecat.purchases.hybridcommon.setLogHandlerWithOnResult
import com.revenuecat.purchases.hybridcommon.setLogLevel
import com.revenuecat.purchases.hybridcommon.setProxyURLString
import com.revenuecat.purchases.hybridcommon.setPurchasesAreCompletedBy
import com.revenuecat.purchases.hybridcommon.showInAppMessagesIfNeeded
import com.revenuecat.purchases.hybridcommon.syncPurchases
import com.revenuecat.purchases.models.InAppMessageType

@Suppress("unused", "DEPRECATION", "LongParameterList", "UNUSED_VARIABLE")
private class CommonApiTests {
    fun checkSetAllowSharingAppStoreAccount(allow: Boolean) {
        setAllowSharingAppStoreAccount(allow)
    }

    fun checkGetOfferings(onResult: OnResult) {
        getOfferings(onResult)
    }

    fun checkGetProductInfo(
        productIDs: List<String>,
        type: String,
        onResult: OnResultList,
    ) {
        getProductInfo(productIDs, type, onResult)
    }

    fun checkPurchase(
        activity: Activity?,
        options: Map<String, Any?>,
        onResult: OnResult,
    ) {
        purchase(activity, options, onResult)
    }

    fun checkPurchaseProduct(
        activity: Activity?,
        productIdentifier: String,
        type: String,
        googleBasePlanId: String?,
        googleOldProductId: String?,
        googleReplacementMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        presentedOfferingContext: Map<String, Any?>?,
        addOnStoreProducts: List<Map<String, Any?>>?,
        addOnSubscriptionOptions: List<Map<String, Any?>>?,
        addOnPackages: List<Map<String, Any?>>?,
        onResult: OnResult,
    ) {
        purchaseProduct(
            activity,
            productIdentifier,
            type,
            googleBasePlanId,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
        )

        purchaseProduct(
            activity,
            productIdentifier,
            type,
            googleBasePlanId,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
        )

        purchaseProduct(
            activity,
            productIdentifier,
            type,
            googleBasePlanId,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
        )

        purchaseProduct(
            activity,
            productIdentifier,
            type,
            googleBasePlanId,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
            addOnPackages,
        )
    }

    fun checkPurchasePackage(
        activity: Activity?,
        packageIdentifier: String,
        presentedOfferingContext: Map<String, Any?>,
        googleOldProductId: String?,
        googleReplacementMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        onResult: OnResult,
        addOnStoreProducts: List<Map<String, Any?>>?,
        addOnSubscriptionOptions: List<Map<String, Any?>>?,
        addOnPackages: List<Map<String, Any?>>?,
    ) {
        purchasePackage(
            activity,
            packageIdentifier,
            presentedOfferingContext,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            onResult,
        )

        purchasePackage(
            activity,
            packageIdentifier,
            presentedOfferingContext,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            onResult,
            addOnStoreProducts,
        )

        purchasePackage(
            activity,
            packageIdentifier,
            presentedOfferingContext,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
        )

        purchasePackage(
            activity,
            packageIdentifier,
            presentedOfferingContext,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
            addOnPackages,
        )
    }

    fun checkPurchaseSubscriptionOption(
        activity: Activity?,
        productIdentifier: String,
        optionIdentifier: String,
        googleOldProductId: String?,
        googleReplacementMode: Int?,
        googleIsPersonalizedPrice: Boolean?,
        presentedOfferingContext: Map<String, Any?>?,
        onResult: OnResult,
        addOnStoreProducts: List<Map<String, Any?>>?,
        addOnSubscriptionOptions: List<Map<String, Any?>>?,
        addOnPackages: List<Map<String, Any?>>?,
    ) {
        purchaseSubscriptionOption(
            activity,
            productIdentifier,
            optionIdentifier,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
        )

        purchaseSubscriptionOption(
            activity,
            productIdentifier,
            optionIdentifier,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
        )

        purchaseSubscriptionOption(
            activity,
            productIdentifier,
            optionIdentifier,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
        )

        purchaseSubscriptionOption(
            activity,
            productIdentifier,
            optionIdentifier,
            googleOldProductId,
            googleReplacementMode,
            googleIsPersonalizedPrice,
            presentedOfferingContext,
            onResult,
            addOnStoreProducts,
            addOnSubscriptionOptions,
            addOnPackages,
        )
    }

    fun checkGetAppUserId() {
        val appUserId: String = getAppUserID()
    }

    fun checkStorefrontCountryCode(callback: (Map<String, Any>?) -> Unit) {
        getStorefront(callback)
    }

    fun checkRestorePurchases(onResult: OnResult) {
        restorePurchases(onResult)
    }

    fun checkLogIn(
        appUserID: String,
        onResult: OnResult,
    ) {
        logIn(appUserID, onResult)
    }

    fun checkLogOut(onResult: OnResult) {
        logOut(onResult)
    }

    fun checkSetDebugLogsEnabled(enabled: Boolean) {
        setDebugLogsEnabled(enabled)
    }

    fun checkSetLogLevel(level: String) {
        setLogLevel(level)
    }

    fun checkSetLogHandler(callback: (logDetails: Map<String, String>) -> Unit) {
        setLogHandler(callback)
    }

    @Suppress("EmptyFunctionBlock")
    fun checkSetLogHandlerWithOnResult() {
        setLogHandlerWithOnResult(object : OnResult {
            override fun onReceived(map: Map<String, *>) {}
            override fun onError(errorContainer: ErrorContainer) {}
        })
    }

    fun checkSetProxyURLString(proxyURLString: String?) {
        setProxyURLString(proxyURLString)
    }

    fun checkGetProxyURLString() {
        val proxyURLString: String? = getProxyURLString()
    }

    fun checkGetCustomerInfo(onResult: OnResult) {
        getCustomerInfo(onResult)
    }

    fun checkSyncPurchases(onResult: OnResult) {
        syncPurchases()
        syncPurchases(onResult)
    }

    fun checkIsAnonymous() {
        val isAnonymous: Boolean = isAnonymous()
    }

    fun checkSetPurchasesAreCompletedBy(purchasesAreCompletedBy: String) {
        setPurchasesAreCompletedBy(purchasesAreCompletedBy)
    }

    fun checkCheckTrialOrIntroductoryPriceEligibility(productIdentifiers: List<String>) {
        val result: Map<String, Map<String, Any>> = checkTrialOrIntroductoryPriceEligibility(
            productIdentifiers,
        )
    }

    fun checkInvalidateCustomerInfoCache() {
        invalidateCustomerInfoCache()
    }

    fun checkCanMakePayments(
        context: Context,
        features: List<Int>,
        onResult: OnResultAny<Boolean>,
    ) {
        canMakePayments(context, features, onResult)
    }

    fun checkShowInAppMessagesIfNeeded(activity: Activity?) {
        showInAppMessagesIfNeeded(activity)
        showInAppMessagesIfNeeded(activity, listOf(InAppMessageType.BILLING_ISSUES))
        showInAppMessagesIfNeeded(activity, null)
    }

    fun checkConfigure(
        context: Context,
        apiKey: String,
        appUserID: String?,
        purchasesAreCompletedBy: String?,
        platformInfo: PlatformInfo,
        store: Store,
        dangerousSettings: DangerousSettings,
        shouldShowInAppMessagesAutomatically: Boolean?,
        verificationMode: String?,
        pendingTransactionsForPrepaidPlansEnabled: Boolean?,
        diagnosticsEnabled: Boolean?,
        automaticDeviceIdentifierCollectionEnabled: Boolean?,
        preferredLocale: String?,
    ) {
        configure(context, apiKey, appUserID, purchasesAreCompletedBy, platformInfo)
        configure(context, apiKey, appUserID, purchasesAreCompletedBy, platformInfo, store, dangerousSettings)
        configure(
            context,
            apiKey,
            appUserID,
            purchasesAreCompletedBy,
            platformInfo,
            store,
            dangerousSettings,
            shouldShowInAppMessagesAutomatically,
        )
        configure(
            context,
            apiKey,
            appUserID,
            purchasesAreCompletedBy,
            platformInfo,
            store,
            dangerousSettings,
            shouldShowInAppMessagesAutomatically,
            verificationMode,
            pendingTransactionsForPrepaidPlansEnabled = pendingTransactionsForPrepaidPlansEnabled,
            diagnosticsEnabled = diagnosticsEnabled,
        )
        configure(
            context,
            apiKey,
            appUserID,
            purchasesAreCompletedBy,
            platformInfo,
            store,
            dangerousSettings,
            shouldShowInAppMessagesAutomatically,
            verificationMode,
            pendingTransactionsForPrepaidPlansEnabled = pendingTransactionsForPrepaidPlansEnabled,
            diagnosticsEnabled = diagnosticsEnabled,
            automaticDeviceIdentifierCollectionEnabled = automaticDeviceIdentifierCollectionEnabled,
        )
        configure(
            context,
            apiKey,
            appUserID,
            purchasesAreCompletedBy,
            platformInfo,
            store,
            dangerousSettings,
            shouldShowInAppMessagesAutomatically,
            verificationMode,
            pendingTransactionsForPrepaidPlansEnabled = pendingTransactionsForPrepaidPlansEnabled,
            diagnosticsEnabled = diagnosticsEnabled,
            automaticDeviceIdentifierCollectionEnabled = automaticDeviceIdentifierCollectionEnabled,
            preferredLocale = preferredLocale,
        )
    }

    fun checkGetPromotionalOffer() {
        val errorContainer: ErrorContainer = getPromotionalOffer()
    }

    fun checkErrorContainer(code: Int, message: String, info: Map<String, Any?>) {
        val errorContainer = ErrorContainer(code, message, info)
        val storedCode = errorContainer.code
        val storedMessage = errorContainer.message
        val storedInfo = errorContainer.info
    }

    private fun checkGetVirtualCurrencies(onResult: OnResult) {
        getVirtualCurrencies(onResult)
    }

    private fun checkInvalidateVirtualCurrenciesCache() {
        invalidateVirtualCurrenciesCache()
    }

    private fun checkGetCachedVirtualCurrencies() {
        val cachedVirtualCurrencies: Map<String, Any?>? = getCachedVirtualCurrencies()
    }

    private fun checkOverridePreferredLocale(locale: String?) {
        overridePreferredLocale(locale)
    }
}
